core/src/main/java/com/devonfw/tools/solicitor/licensetexts/StrategyLicenseUrlGuesser.java

```diff
@@ -154,4 +168,44 @@ public GuessedLicenseUrlContent getContentForUri(String uri) {
     return new GuessedLicenseUrlContent(guessedUrl, auditLogBuilder.toString());
   }
 
+   //tries to get github license file location based of vsc-link 
+   public String githubAPILicenseUrl(String link, String token) {
+	 
+     String result = "";
+     if (link.contains("github.com")) {
+       if (link.endsWith(".git")) {
+         link = link.substring(0, link.length() - 4);
+     }
+     link = link.replace("git://", "https://");
+     link = link.replace("ssh://", "https://");
+     link = link.replace("git@", "");
+     if(!link.contains("api.github.com")) {
+       link = link.replace("github.com/", "api.github.com/repos/");
+       link = link.concat("/license");
+     }
+   
+     String command = "curl -H \"Accept: application/vnd.github+json\" -H \"Authorization: token "+ token + "\" -i " + link;
+     ProcessBuilder processBuilder = new ProcessBuilder(command.split(" "));
```

Comment:
@pgarus97 : Lift is marking this as a security issue (possible command injection). Independently of this security issue: Why do we spawn a separate operating system process just to do a simple HTTP GET here? To be independent of the underlying operating system (curl command) we should use Java/Spring functionality to do the web call. This can be done e.g. using RestTemplate (which is or will soon be deprecated) or (preferably) the newer WebClient.

Dependency needed for RestTemplate:
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-web</artifactId>

and for WebClient:

    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-webflux</artifactId>